File: competitive-programming/leet-code/524/findLongestWord.js
var findLongestWord = function (s, dictionary) {
    const isSubsequence = (word) => {
        let i = 0;
        for (let c of s) {
            if (c === word[i]) i++;
            if (i === word.length) return true;
        }
        return false;
    };

    dictionary.sort((a, b) => {
        if (b.length !== a.length) return b.length - a.length;
        return a.localeCompare(b);
    });

    for (let word of dictionary) {
        if (isSubsequence(word)) return word;
    }

    return "";
};

File: competitive-programming/leet-code/524/find_longest_word.py
from typing import List

def find_longest_word(s: str, dictionary: List[str]) -> str:
    def is_subsequence(word: str) -> bool:
        i = 0
        for c in s:
            if c == word[i]:
                i += 1
            if i == len(word):
                return True
        return False

    dictionary.sort(key=lambda x: (-len(x), x))

    for word in dictionary:
        if is_subsequence(word):
            return word

    return ""